detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsLambdaDetector.ts

```diff
@@ -18,20 +18,20 @@ import {
   Detector,
   Resource,
   ResourceDetectionConfig,
-} from '@opentelemetry/resources';
+} from "@opentelemetry/resources";
 import {
   CloudProviderValues,
   CloudPlatformValues,
   SemanticResourceAttributes,
-} from '@opentelemetry/semantic-conventions';
+} from "@opentelemetry/semantic-conventions";
 
 /**
  * The AwsLambdaDetector can be used to detect if a process is running in AWS Lambda
  * and return a {@link Resource} populated with data about the environment.
  * Returns an empty Resource if detection fails.
  */
 export class AwsLambdaDetector implements Detector {
-  async detect(_config?: ResourceDetectionConfig): Promise<Resource> {
+  detectSync(_config?: ResourceDetectionConfig): Resource {
```

Comment:
If we add this option for sync detection, we should probably also document it in the README so users are aware of it and how to leverage it.